aws/sdk/build.gradle.kts

```diff
@@ -452,6 +471,50 @@ tasks.register("sdkTest") {
     finalizedBy(Cargo.CLIPPY.toString, Cargo.TEST.toString, Cargo.DOCS.toString)
 }
 
+//Tasks for generating individual Cargo.lock files
+fun Project.registerLockfileGeneration(
+    dir: File,
+    name: String,
+): String {
+    val taskName = "generate${name}Lockfile"
+    tasks.register<Exec>(taskName) {
+        workingDir(dir)
+        environment("RUSTFLAGS", "--cfg aws_sdk_unstable")
+        commandLine("cargo", "generate-lockfile")
+    }
+
+    return taskName
+}
+
+val registerAwsConfigTaskName = registerLockfileGeneration(awsConfigPath, "AwsConfig")
+val registerAwsRustRuntimeTaskName = registerLockfileGeneration(awsRustRuntimePath, "AwsRustRuntime")
+val registerRustRuntimeTaskName = registerLockfileGeneration(rustRuntimePath, "RustRuntime")
+
+tasks.register<Exec>("generateAwsSdkRustLockfile") {
+    val sdkRustPath: String =
+        properties.get("aws-sdk-rust-path") ?: throw Exception("A -Paws-sdk-rust-path argument must be specified")
+    workingDir(sdkRustPath)
+    environment("RUSTFLAGS", "--cfg aws_sdk_unstable")
+    commandLine("cargo", "generate-lockfile")
+    copy {
+        from(sdkRustPath + "/Cargo.lock")
+        into(rootProject.projectDir.resolve("aws/sdk"))
+    }
+}
+
+//Parent task to generate all the Cargo.lock files
+tasks.register("generateAllLockfiles") {
```

Comment:
I was pondering if this should be named something like `updateAllLockfiles` 🤔 , I guess we call `generate-lockfile` from `cargo` so this is in line with that

Reply:
I kept generate to align with the cargo command, but it will always be update going forward, so I'm not opposed to the change if you have a strong opinion